framework/derive/src/format/format_version.rs

```diff
@@ -0,0 +1,37 @@
+use proc_macro::{quote, Literal};
+
+use crate::format::format_tokenize;
+
+pub fn format_version(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
```

Comment:
I would rename this to `version_triple`, or `version_tuple`,  or `semver_tuple`, whichever you prefer. Same name for the file.